extensions/cache/runtime/src/main/java/io/quarkus/cache/CompositeCacheKey.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.cache;
+
+import java.util.Arrays;
+
+/**
+ * A composite cache key is used by the annotations caching API when a method annotated with {@link CacheResult} or
+ * {@link CacheInvalidate} is invoked and when the cache key is composed of several of the method arguments (annotated with
+ * {@link CacheKey} or not). This class can also be used with the programmatic caching API.
+ */
+public class CompositeCacheKey {
+
+    private final Object[] keyElements;
+
+    /**
+     * Constructor.
+     * 
+     * @param keyElements key elements
+     * @throws IllegalArgumentException if less than two key elements are provided
+     */
+    public CompositeCacheKey(Object... keyElements) {
+        if (keyElements.length < 2) {
+            throw new IllegalArgumentException(
```

Comment:
We can wait but I imagine "generic" usages of the API will complaina bout this check

Reply:
I changed the check, one element is required instead of two now.